modules/reactor/src/meshgenerators/AzimuthalBlockSplitGenerator.C

```diff
@@ -454,3 +556,21 @@ AzimuthalBlockSplitGenerator::nodeModifier(
     }
   }
 }
+
+Point
+AzimuthalBlockSplitGenerator::midPointCorrector(const Point vertex_0,
+                                                const Point vertex_1,
+                                                const Real max_radius,
+                                                const Real rad_tol)
+{
+  // Check if two vertices have the same radius
+  const Real r_vertex_0 = std::sqrt(vertex_0(0) * vertex_0(0) + vertex_0(1) * vertex_0(1));
+  const Real r_vertex_1 = std::sqrt(vertex_1(0) * vertex_1(0) + vertex_1(1) * vertex_1(1));
+  // If both vertices have the same radius and they are located in the circular region,
```

Comment:
we dont really know that they are located in the circular region, same radius could be coincidental?
is this meant for all the elements touching the cutting plane and on the outer rim of fuel pin / core outer zone?

Reply:
We checked the max radius of the circular regions before and use it to tell whether the vertices belong to circular regions here. The radius comparison between the radii here is used to tell the orientation of the edge determined by the two vertices (tangential or normal).